### ISSUE TITLE
Feature: AppArmor Profile

### ISSUE THREAD

**Issue by eslerm** [ubuntu/authd#199]
### Is there an existing request for this feature?

- [X] I have searched the existing issues and found none that matched mine

### Describe the feature

Please add an AppArmor profile. This will require a profile in main and it would be great if Authd could provide this directly. 

[Please take care to confine D-Bus](https://bugs.launchpad.net/ubuntu/+source/dbus-broker/+bug/2015538/comments/6).

Security is happy to review.

### Environment
* authd version: please run //TODO
* Distribution: (**NAME** in `/etc/os-release`)
* Distribution version: (**VERSION_ID** on `/etc/os-release`):

### Log files
Please redact/remove sensitive information:
```raw
authd logs can be found in //TODO
```

### Application settings
Please redact/remove sensitive information:
```raw
You can get the configuration file from //TODO
```

### Double check your logs

- [X] I have redacted any sensitive information from the logs

**Comment by didrocks** [comment]
Thanks for the hint! We would be happy to this, however we do have a few questions:
- authd is only a dbus client, not the daemon, is it relevant for an apparmor profile?
- do you have example of dbus confinement in such a case (client?).

Is the systemd service confinement not enough already? I’m afraid we don’t have the Apparmor expertise on our team, and so, we would like to have some guidance/help by the security team to unblock us here.

**Comment by eslerm** [comment]
I'm reaching out for AppArmor expertise myself. I'll try to come back with some concrete examples from the AppArmor team.

AppArmor.d may have some useful examples, like [polkitd](https://github.com/roddhjav/apparmor.d/blob/c006371e5beb653adf9678411e0dbf8cbb583fb7/apparmor.d/groups/freedesktop/polkitd) or profiles that use features like nss.

Just to be clear, lack of an AppArmor profile will not block authd's MIR, but will be needed in 24.04.

**Comment by didrocks** [comment]
Thanks for reaching out on guidance! We will look at examples too (but I’m unsure in particular about the dbus part as we are only client there).

Please keep us posted once you have concrete profile examples we can shamelessly steal from :) That sounds like a perfect post Feature Freeze task.

**Comment by gegarcia** [comment]
It is not unusual to have AppArmor profiles for dbus clients, since dbus daemon verifies that bus communication is allowed by an AppArmor policy. Our recommendation is to use the AppArmor tools like aa-genprof and aa-logprof to exercise authd's code and construct the AppArmor policy that way. After  you have a working policy, there might be still some denials that might be caught by corner cases, so while the policy is not mature enough, you could have the profile in complain mode. Complain mode allows what is not in policy, but it logs it in your system logs.

**Comment by stendler-mlpa** [comment]
Not sure if this warrants a separate issue, so I report this here for now: the apparmor profile of fusermount3 needs adjustment for authd as well -- similar to the issue with [fusermount3/flatpak on ubuntu 25.10](https://bugs.launchpad.net/ubuntu/+source/flatpak/+bug/2122161) and [polkit/unix_chkpwd + authd](https://launchpadlibrarian.net/812123851/apparmor_5.0.0~alpha1-0ubuntu3_5.0.0~alpha1-0ubuntu4.diff.gz).

Current workaround is to add `/etc/apparmor.d/local/fusermount3` with:

```
# Rules required if authd is used
@{run}/authd.sock rw,
```
